src/general/buffer.jl

```diff
@@ -4,7 +4,8 @@ struct SystemBuffer{V}
     buffer_size     :: Int
 
     function SystemBuffer(active_size, buffer_size::Integer)
-        active_particle = vcat(trues(active_size), falses(buffer_size))
+        # We cannot use a `BitVector` here, as writing to a `BitVector` is not thread-safe
+        active_particle = vcat(fill(true, active_size), fill(false, buffer_size))
```

Comment:
please change `active_particle :: BitVector` to `active_particle :: Vector{Bool}` above